### PR TITLE
making corrections to the P4₁₆ spec. based on reading version 1.2.2

### PR DESCRIPTION
**Pull request by Abe149**

### CODE REVIEW

**Comment by jnfoster** [comment]
Realize this is a WIP, but thanks in advance for all this work! Great stuff.

File: p4-16/spec/P4-16-spec.mdk [hunk 1]
```diff
@@ -1588,7 +1588,7 @@ the same result):
       architecture.
 6.	On function return, the temporaries that correspond to `out`
       or `inout` arguments are copied in order from left to right
-      into the l-values saved in Step 3.
+      into the l-values saved in Step 2.
```
Comment (mbudiu-vmw) [inline p4-16/spec/P4-16-spec.mdk:1591]: Step 3 talks about saving the l-values.

**Review by mbudiu-vmw — CHANGES_REQUESTED** [review]

File: p4-16/spec/P4-16-spec.mdk (hunk 1 above)
Comment (Abe149) [reply p4-16/spec/P4-16-spec.mdk:1591]: You are correct.  I don`t know how I got this wrong.  Possibly I got distracted in the middle of reading the list of steps.  I will revert this change.

Comment (Abe149) [reply p4-16/spec/P4-16-spec.mdk:1591]:
```
$ git show 043ade01c924f8c0eba834495ee920dd53062a6c
commit 043ade01c924f8c0eba834495ee920dd53062a6c
Author: Abe Skolnik <abe.skolnik@intel.com>
Date:   Mon Apr 4 18:34:40 2022 -0700

    in {repo.}/“p4-spec/p4-16/P4-16-spec.mdk”: fixed an off-by-one reference to another step in a list of steps (in the subsection on the calling convention)

diff --git a/p4-16/spec/P4-16-spec.mdk b/p4-16/spec/P4-16-spec.mdk
index c08b573..7462d43 100644
--- a/p4-16/spec/P4-16-spec.mdk
+++ b/p4-16/spec/P4-16-spec.mdk
@@ -1588,7 +1588,7 @@ the same result):
       architecture.
 6.     On function return, the temporaries that correspond to `out`
       or `inout` arguments are copied in order from left to right
-      into the l-values saved in Step 3.
+      into the l-values saved in Step 2.

 According to this algorithm, the previous function call is equivalent
 to the following sequence of statements:
```

**Review by mbudiu-vmw — APPROVED** [review]

**Review by mbudiu-vmw — APPROVED** [review]

**Comment by mbudiu-vmw** [comment]
Can we merge this? You can submit more fixes in a separate PR. Makes for easier reviews.

**Comment by Abe149** [comment]
@mbudiu-vmw  Should I squash it, or should I preserve my commit granularity?

**Comment by Abe149** [comment]
@mbudiu-vmw Should I squash it, or should I preserve my commit granularity?

**Comment by mbudiu-vmw** [comment]
This is fine.

**Comment by Abe149** [comment]
Fixed the [mostly-fake] conflict.

**Pull request merged**